src/main/java/org/prebid/server/cache/CoreCacheService.java

```diff
@@ -553,4 +585,44 @@ private BidCacheRequest toBidCacheRequest(List<CachedCreative> cachedCreatives)
                 .map(CachedCreative::getPayload)
                 .toList());
     }
+
+    private String resolveCacheKey(String accountId) {
+        return resolveCacheKey(accountId, null);
+    }
+
+    // hbCacheId is accepted here to have a backwards-compatibility with video category mapping
+    private String resolveCacheKey(String accountId, String hbCacheId) {
+        if (!appendTraceInfoToCacheId) {
+            return hbCacheId;
+        }
+
+        // no need to have additional separator if datacenter name won't be added
+        final boolean isDatacenterNamePopulated = StringUtils.isNotBlank(datacenterRegion);
+        final int separatorCount = isDatacenterNamePopulated ? 2 : 1;
+        final int accountIdLength = accountId.length();
+        final int traceInfoLength = isDatacenterNamePopulated
+                ? accountIdLength + datacenterRegion.length() + separatorCount
+                : accountIdLength + separatorCount;
+
+        final String cacheKey = hbCacheId != null ? hbCacheId : idGenerator.generateId();
+        if (cacheKey != null && traceInfoLength < cacheKey.length()) {
```

Comment:
lets invert if and return hbCacheId here. So that big chunk of the code will be without unnecessary indentation.